packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -563,12 +564,23 @@ class Chromium {
   }
 }
 
-// TODO(andrewkolos): Remove when https://github.com/dart-lang/sdk/issues/56566
-//  is fixed.
 Future<ChromeTab?> getChromeTabGuarded(
   ChromeConnection chromeConnection,
   bool Function(ChromeTab tab) accept, {
   Duration? retryFor,
-}) {
-  return asyncGuard(() => chromeConnection.getTab(accept, retryFor: retryFor));
+  void Function(Object error, StackTrace stackTrace)? onError,
+}) async {
+  try {
+    return asyncGuard(() => chromeConnection.getTab(accept, retryFor: retryFor));
+  } on IOException catch (error, stackTrace) {
+    if (onError != null) {
+      onError(error, stackTrace);
+    }
+    return null;
+  } on StateError catch (error, stackTrace) {
```

Comment:
can you add a comment why we're catching StateError?

Reply:
Addressed in [refactor: add clarifying docs](https://github.com/flutter/flutter/pull/154366/commits/9fd3e846b7d2cb8761637839cbca769d38e3f0f3)